openmc/openmoc_compatible.py

```diff
@@ -0,0 +1,825 @@
+import copy
+import operator
+
+import numpy as np
+
+import openmoc
+import openmc
+import openmc.checkvalue as cv
+
+
+# A dictionary of all OpenMC Materials created
+# Keys   - Material IDs
+# Values - Materials
+OPENMC_MATERIALS = {}
+
+# A dictionary of all OpenMOC Materials created
+# Keys   - Material IDs
+# Values - Materials
+OPENMOC_MATERIALS = {}
+
+# A dictionary of all OpenMC Surfaces created
+# Keys   - Surface IDs
+# Values - Surfaces
+OPENMC_SURFACES = {}
+
+# A dictionary of all OpenMOC Surfaces created
+# Keys   - Surface IDs
+# Values - Surfaces
+OPENMOC_SURFACES = {}
+
+# A dictionary of all OpenMC Cells created
+# Keys   - Cell IDs
+# Values - Cells
+OPENMC_CELLS = {}
+
+# A dictionary of all OpenMOC Cells created
+# Keys   - Cell IDs
+# Values - Cells
+OPENMOC_CELLS = {}
+
+# A dictionary of all OpenMC Universes created
+# Keys   - Universes IDs
+# Values - Universes
+OPENMC_UNIVERSES = {}
+
+# A dictionary of all OpenMOC Universes created
+# Keys   - Universes IDs
+# Values - Universes
+OPENMOC_UNIVERSES = {}
+
+# A dictionary of all OpenMC Lattices created
+# Keys   - Lattice IDs
+# Values - Lattices
+OPENMC_LATTICES = {}
+
+# A dictionary of all OpenMOC Lattices created
+# Keys   - Lattice IDs
+# Values - Lattices
+OPENMOC_LATTICES = {}
```

Comment:
Typically not a good idea to keep state in a global variable like this as you can easily run into race conditions. Is this something that can be put into an instance or class method?